secure-api-gateway-ig-extensions/README.md

```diff
@@ -0,0 +1,95 @@
+| maxCacheEntries          | The maximum number of values that can be stored in the cache                                                                                                           | 100                                                    |
+| expireAfterWriteDuration | Values will be removed from the cache when this time elapses after the value was last written.<br/>This is a String representation of org.forgerock.util.time.Duration | "5 minutes"                                            |
+| handler                  | The org.forgerock.openig.handler.ClientHandler instance to use to send the HTTP requests to fetch the JWKS data                                                        | "ClientHandler", which is the default IG ClientHandler |
+
+### No Caching
+```
+{
+      "name": "OBJwkSetService",
+      "type": "RestJwkSetService"
+}
+```
+
+| config option | description                                                                                                                                        | default                                             |
+|---------------|----------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------|
+| handler       | The org.forgerock.openig.handler.ClientHandler instance to use to send the HTTP requests to fetch the JWKS data                                    | "ClientHandler", which is the default ClientHandler |
+
+### Route configuration
+The [ProcessDetachedSig](../config/7.1.0/securebanking/ig/scripts/groovy/ProcessDetachedSig.groovy) filter has a dependency on a JwkSetService object in order to get JWKS data to verify the `x-jws-signature` header sent in OBIE API requests. 
+
+Routes which use the ProcessDetachedSig filter must ensure that they configure the filter arg: `"jwkSetService": "${heap['OBJwkSetService']}"`
+
+## Extensions
+Clients can use an alternative caching library by writing an adaptor class which implements the [com.forgerock.sapi.gateway.jwks.cache.Cache](src/main/java/com/forgerock/sapi/gateway/jwks/cache/Cache.java) interface. 
+
+The contents of package [com.forgerock.sapi.gateway.jwks.cache.caffeine](src/main/java/com/forgerock/sapi/gateway/jwks/cache/caffeine/) demonstrate how this can be done. In this package there is: a CaffeineCache adaptor class, CaffeineCachingJwkSetService (which extends CachingJwkSetService) and a heaplet object which is used to construct and configure the caffeine caching. 
+
+# FAPI Support
+This module contains classes and Filters which can help with achieving [FAPI](https://openid.net/specs/openid-financial-api-part-2-1_0.html) conformance.
```

Comment:
Part 2? Maybe some more explanation here. Part 1 plus bits of Part 2?

Reply:
That's the link to the advanced spec (Part 2). I'll add some more info to make it clearer.